sift/src/main/java/siftscience/android/AppStateCollector.java

```diff
@@ -284,17 +284,22 @@ public void onSuccess(LocationSettingsResponse locationSettingsResponse) {
                     .addOnFailureListener(new OnFailureListener() {
                         @Override
                         public void onFailure(@NonNull Exception e) {
-                            int statusCode = ((ApiException) e).getStatusCode();
-                            switch (statusCode) {
-                                case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
-                                    Log.i(TAG, "Location settings are not satisfied. " +
-                                            "Try to attempt upgrade location settings");
-                                    break;
-                                case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
-                                    Log.i(TAG, "Location settings are inadequate, and cannot be " +
-                                            "fixed here. Fix in Settings.");
-                                    mRequestingLocationUpdates = false;
+                            if (e instanceof ApiException) {
+                                int statusCode = ((ApiException) e).getStatusCode();
+                                switch (statusCode) {
+                                    case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
+                                        Log.i(TAG, "Location settings are not satisfied. " +
+                                                "Try to attempt upgrade location settings");
+                                        break;
+                                    case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+                                        Log.i(TAG, "Location settings are inadequate, and cannot be " +
+                                                "fixed here. Fix in Settings.");
+                                }
+                            } else {
+                                // A different, unknown type of error occurred.
+                                Log.d(TAG, "ERROR! " + e.getMessage());
                             }
+                            mRequestingLocationUpdates = false;
```

Comment:
Shouldn't this line be moved to line 297, to match the same logic as the previous version?

Reply:
In both cases that line is required, that is why it was moved outside.

Reply:
With the old logic I believe that mRequestingLocationUpdates would only equal false when the statusCode = LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE and not in both cases, right?
```suggestion
switch (statusCode) {
    case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
        Log.i(TAG, "Location settings are not satisfied. " +
                "Try to attempt upgrade location settings");
        break;
    case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
        Log.i(TAG, "Location settings are inadequate, and cannot be " +
                "fixed here. Fix in Settings.");
        mRequestingLocationUpdates = false;
```